apps/src/netsim/NetSimLogPanel.js

```diff
@@ -188,8 +188,19 @@ NetSimLogPanel.prototype.onClearButtonPress_ = function () {
 
 /**
  * Put a message into the log.
+ * @param {string} packetBinary
+ * @param {number} packetID
  */
-NetSimLogPanel.prototype.log = function (packetBinary) {
+NetSimLogPanel.prototype.log = function (packetBinary, packetID) {
+
+  var packetAlreadyInLog = this.packets_.some(function (packet) {
+    return packet.packetID_ == packetID;
```

Comment:
Also, I have no idea if we already violate this all over the place in this file :grimacing:.